python/src/uagents/agent.py

```diff
@@ -996,11 +1061,13 @@ async def run_async(self):
 
         """
         await self.setup()
+
+        tasks = [self._server.serve()]
```

Comment:
Should we check for any registered REST handlers before running the server? Seems like this could save resources for a mailbox agent that has opted out of handling any rest endpoints, but not sure how significant.